### PR TITLE
[common] revert started flag when service already stopped

### PR DESCRIPTION
**Pull request by melekes**
also, return ErrNotStarted when trying to stop a not-running service

### CODE REVIEW

File: libs/common/service.go [hunk 1]
```diff
@@ -124,6 +131,8 @@ func (bs *BaseService) Start() error {
 	if atomic.CompareAndSwapUint32(&bs.started, 0, 1) {
 		if atomic.LoadUint32(&bs.stopped) == 1 {
 			bs.Logger.Error(fmt.Sprintf("Not starting %v -- already stopped", bs.name), "impl", bs.impl)
+			// revert flag
+			atomic.StoreUint32(&bs.started, 0)
```
Comment (ValarDragon) [inline libs/common/service.go]: I'm confused as to why we have separate started and stopped flags. Shouldn't `started = !stopped`?

Comment (melekes) [reply libs/common/service.go]: There was a discussion about simplifying `Service` and @jaekwon made a PR https://github.com/tendermint/tmlibs/pull/114. Not sure why we did not merge it.

**Comment by ebuchman** [comment]
What motivated this ?

**Comment by melekes** [comment]
https://github.com/tendermint/tendermint/issues/2258

```
func(c *rpcClient.WSClient) error {
		if c.IsRunning() {
			c.Stop() // panics on close c.ResponsesCh when not started (=nil)
		}
```

We should not panic if somebody's trying to Stop a not-started WSClient.

**Comment by ValarDragon** [comment]
This PR seems good to me, but I'd prefer if we removed`started` and `stopped` and just had `running`.

**Comment by melekes** [comment]
How about we merge this and bring https://github.com/tendermint/tmlibs/pull/114 in?

**Comment by jaekwon** [comment]
I think this is fine for now.
The reason why it's still split is to consider an API for restarting services... restarting may require different semantics than starting.  Currently these services are run-once-only, so you can't restart a stopped process, and that may be what we want ultimately, and even if not, it's better to have a restricted API and to expand it further.  Given the way we use services currently (that there is no restart-ability), we may end up masking bugs by allowing them to be restarted after being stopped.

If we want to allow for services to restart, it may be better to require an explicit "restarting".  If in the future we know that (a) we want the ability for services to be started and stopped and started again and (b) we don't want to require explicit "restart" calls, then it would make sense to merge start/stop, but I prefer the way things are for the reasons above.

**Pull request merged**